WordPress/gutenberg
# Lodash: Refactor away from `_.mergeWith()`

## What?
This PR removes the last usage of Lodash's `_.mergeWith()` and deprecates the function.

~Blocked by #50637 and should be rebased after it lands.~

## Why?

Lodash is known to unnecessarily inflate the bundle size of packages, and in most cases, it can be replaced with native language functionality. See these for more information and rationale:

* https://github.com/WordPress/gutenberg/issues/16938#issuecomment-602837246
* https://github.com/WordPress/gutenberg/issues/17025
* https://github.com/WordPress/gutenberg/issues/39495 

## How?

Just like we've been doing in other places, we're replacing `_.mergeWith()` with the `deepmerge` package. 

To ensure we're not merging arrays, we're adding a `isMergeableObject: isPlainObject` setting that will only merge objects and fully override arrays to preserve the pre-existing behavior.

## Testing Instructions

* Verify global styles continue to work and that user-provided styles override the global ones.
* Verify all checks are green.
